### PR TITLE
Block cmdlets for AADDS or AADKERB

### PR DESCRIPTION
**Pull request by clin-ms**
Block cmdlets for AADDS or AADKERB

### CODE REVIEW

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 1]
```diff
@@ -219,6 +219,61 @@ function Assert-IsDomainJoined {
     }
 }
 
+function Assert-IsNotAADDS {
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:222]: Better name it Assert-IsAD or Assert-IsNativeAD. It should fail if the storage account is not configured for AD.

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 2]
```diff
@@ -219,6 +219,61 @@ function Assert-IsDomainJoined {
...
+            -ErrorAction Stop
+    }
+
+    if (
+        $null -ne $StorageAccount.AzureFilesIdentityBasedAuth -and`
+        $null -ne $StorageAccount.AzureFilesIdentityBasedAuth.DirectoryServiceOptions -and `
+        !(
+            "AADDS" -eq $StorageAccount.AzureFilesIdentityBasedAuth.DirectoryServiceOptions -or `
+            "AADKERB" -eq $StorageAccount.AzureFilesIdentityBasedAuth.DirectoryServiceOptions
+        )
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:263]: Should just check whether it is AD.

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 1 above)
Comment (clin-ms) [reply AzFilesHybrid/AzFilesHybrid.psm1]: Resolved

File: AzFilesHybrid/AzFilesHybrid.psm1 (hunk 2 above)
Comment (clin-ms) [reply AzFilesHybrid/AzFilesHybrid.psm1]: Resolved

File: AzFilesHybrid/AzFilesHybrid.psm1 [hunk 3]
```diff
@@ -219,6 +219,58 @@ function Assert-IsDomainJoined {
     }
 }
 
+function Assert_IsNativeAD {
```
Comment (YunZhou-MS) [inline AzFilesHybrid/AzFilesHybrid.psm1:222]: Use - to replace  _ in cmdlet name

**Review by YunZhou-MS — APPROVED** [review]

**Review by ledavies — APPROVED** [review]

**Pull request merged**